x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java

```diff
@@ -835,6 +887,37 @@ public Builder validate(boolean forCreation) {
                         validationException
                     );
                 }
+
+                // packaged model validation
+                validationException = checkIllegalSetting(modelPackageConfig, MODEL_PACKAGE.getPreferredName(), validationException);
+
+                if (packagedModel) {
+                    validationException = checkIllegalPackagedModelSetting(
+                        description,
+                        DESCRIPTION.getPreferredName(),
+                        validationException
+                    );
```

Comment:
Is this validation taking place before we know that the packaged model exists?

For example, if I create a model `.my_model` (accidentally typing a dot at the beginning of `my_model`) then will I get an error saying that I cannot have a description on a packaged model? That's a bit cryptic. It would be much nicer to get an error message saying that model IDs beginning with dots are reserved for packaged models and no packaged model called `.my_model` exists.

_However_, since we're in a rush to deliver a working version and this isn't a showstopper, feel free to add a `TODO` and leave it for a followup.